src/templating/native/nativeTemplateEngine.js

```diff
@@ -10,7 +13,16 @@ ko.nativeTemplateEngine.prototype['renderTemplateSource'] = function (templateSo
         templateNodes = templateNodesFunc ? templateSource['nodes']() : null;
 
     if (templateNodes) {
-        return ko.utils.makeArray(templateNodes.cloneNode(true).childNodes);
+        var nodes;
+        if (templateNodes instanceof Array) {
+            nodes = [];
+            for (var i = 0; i < templateNodes.length; ++i) {
```

Comment:
No reason why I used a native for loop vs arrayForEach. Also I am not sure caching the array length really makes much of a difference these days. Even in older browsers it doesn't make much difference with the amount of items we have here. Try http://jsperf.com/cached-array-length and see what you get. My Chrome/Mac was actually marginally faster without caching the array length.